bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env ruby
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'state_abbr'
+require src_dir 'database'
+require_relative('../../dashboard/config/environment')
+require 'cdo/properties'
+require 'json'
+
+DASHBOARD_REPORTING_DB_READONLY = sequel_connect(
+  CDO.dashboard_reporting_db_reader, CDO.dashboard_reporting_db_reader)
+
+UPDATE_BATCH_SIZE = 100
+
+DASHBOARD_DB_NAME = "dashboard_#{Rails.env}"
+
+# List of valid PD courses. If more courses get added, need to update this list and also schema in Pardot. We
+# need to filter here to known courses in the Pardot schema - we can't blindly pass values through
+COURSE_LIST = "'CS Fundamentals','CS in Algebra','CS in Science','CS Principles','Exploring Computer Science','CS Discoveries'"
+
+# values of forms.kind field we care about
+FORM_KINDS = %w(BringToSchool2013 CSEdWeekEvent2013 DistrictPartnerSubmission HelpUs2013 Petition K5OnlineProfessionalDevelopmentPostSurvey)
+
+# information about presence of which forms submitted by a user get recorded in which
+# rollup field with which value
+FORM_INFOS =
+  [
+    {kind: "'CSEdWeekEvent2013'", dest_field: "hoc_organizer_years", dest_value: "'2013'"},
+    {kind: "'HocSignup2014'", dest_field: "hoc_organizer_years", dest_value: "'2014'"},
+    {kind: "'HocSignup2015'", dest_field: "hoc_organizer_years", dest_value: "'2015'"},
+    {kind: "'HocSignup2016'", dest_field: "hoc_organizer_years", dest_value: "'2016'"},
+    {kind: "'Petition'", dest_field: "roles", dest_value: "'Petition Signer'"}
+  ].freeze
+
+def log(s)
+  puts s      # emit to stdout
+  CDO.log.info s   # emit to log file
+end
+
+def log_completion(start)
+  log "   Completed in #{Time.now - start} seconds"
+end
+
+def truncate_contact_rollups
+  start = Time.now
+  log "Truncating contact_rollups"
+  PEGASUS_DB.run "TRUNCATE TABLE contact_rollups"
+  log_completion(start)
+end
+
+def insert_from_dashboard_contacts
+  start = Time.now
+  log "Inserting teacher contacts and IP geo data from dashboard.users"
+  PEGASUS_DB.run "
+    INSERT INTO contact_rollups (email, name, dashboard_user_id, roles, city, state, postal_code, country)
+    SELECT email COLLATE utf8_general_ci, name, users.id, 'Teacher', city, state, postal_code, country FROM #{DASHBOARD_DB_NAME}.users AS users
+    LEFT OUTER JOIN #{DASHBOARD_DB_NAME}.user_geos AS user_geos ON user_geos.user_id = users.id
+    WHERE users.user_type = 'teacher' AND LENGTH(email) > 0 AND user_geos.indexed_at IS NOT NULL
+    ON DUPLICATE KEY UPDATE contact_rollups.name = VALUES(name), contact_rollups.dashboard_user_id = VALUES(dashboard_user_id), roles = VALUES(roles)"
+  log_completion(start)
+end
+
+def insert_from_dashboard_pd_enrollments
+  start = Time.now
+  log "Inserting contacts from dashboard.pd_enrollments"
+  PEGASUS_DB.run "
+    INSERT INTO contact_rollups (email, name)
+    SELECT email, name FROM #{DASHBOARD_DB_NAME}.pd_enrollments AS pd_enrollments
```

Comment:
Looking at the production data, as of this moment `name` is still filled out for all rows (and then parsed into `first_name` and `last_name`). I would like to PR with current code which will work for the moment and then change to `first_name` and `last_name` in a subsequent PR. This is going to require updating my local data and testing.